hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DistributionSetManagementTest.java

```diff
@@ -493,6 +504,16 @@ public void updateDistributionSet() {
         assertThat(ds.isRequiredMigrationStep()).isTrue();
     }
 
+    @Test
+    @Description("Verifies that an exception is thrown when trying to update an invalid distribution set")
+    public void updateInvalidDistributionSet() {
+        final DistributionSet distributionSet = testdataFactory.createAndInvalidateDistributionSet();
+
+        assertThatExceptionOfType(InvalidDistributionSetException.class)
+                .as("Invalid distributionSet should throw an exception").isThrownBy(() -> distributionSetManagement
+                        .update(entityFactory.distributionSet().update(distributionSet.getId()).name("new_name")));
+    }
+
```

Comment:
missing tests
- deleteInvalidDistributionSet
- deleteIncompleteDistributionSet
- updateMetadata throws exception for invalid DS

Reply:
Same here, I think it should be possible to delete an invalidated or incomplete distribution set